src/custom/state/swap/hooks.tsx

```diff
@@ -493,8 +493,11 @@ export function useDetectNativeToken(input?: CurrencyWithAddress, output?: Curre
     // TODO: check the new native currency function
     const native = ETHER.onChain(chainId || DEFAULT_NETWORK_FOR_LISTS)
 
-    const [isNativeIn, isNativeOut] = [input?.currency?.isNative, output?.currency?.isNative]
-    const [isWrappedIn, isWrappedOut] = [input?.currency?.equals(wrappedToken), output?.currency?.equals(wrappedToken)]
+    const [isNativeIn, isNativeOut] = [!!input?.currency?.isNative, !!output?.currency?.isNative]
+    const [isWrappedIn, isWrappedOut] = [
+      !!input?.currency?.equals(wrappedToken),
+      !!output?.currency?.equals(wrappedToken),
+    ]
```

Comment:
Was is failing somewhere?

Reply:
no, just wasn't typed properly as `boolean | undefined` and later always checked as `!!isWrappedIn`